cdap-ui/app/cdap/components/Replicator/Detail/TableScatterPlotGraph/ScatterPlotTooltip.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
+import { ITableMetricsData } from 'components/Replicator/Detail/TableScatterPlotGraph/parser';
+import { numberFormat } from 'components/Replicator/Detail/TableScatterPlotGraph/ScatterPlotTable';
+
+const styles = (theme): StyleRules => {
+  return {
+    tooltip: {
+      position: 'absolute',
+      backgroundColor: theme.palette.white[50],
+      boxShadow: theme.shadows[3],
+      padding: '15px 0',
+      width: '350px',
+    },
+    tooltipHeading: {
+      marginBottom: '25px',
+      padding: '0 15px',
+    },
+    tooltipMetrics: {
+      display: 'grid',
+      gridTemplateColumns: '50% 50%',
+    },
+    metricContent: {
+      padding: '0 15px',
+
+      '&:first-child': {
+        borderRight: `1px solid ${theme.palette.grey[300]}`,
+      },
+
+      '& > div': {
+        display: 'grid',
+        gridTemplateColumns: '50% 50%',
+      },
+    },
+    metric: {
+      textAlign: 'right',
+    },
+  };
+};
+
+interface IScatterPlotTooltipProps extends WithStyles<typeof styles> {
+  tooltip: {
+    top: number;
+    left: number;
+    isOpen: boolean;
+    activeData?: Partial<ITableMetricsData>;
+  };
+}
+
+const ScatterPlotTooltipView: React.FC<IScatterPlotTooltipProps> = ({ classes, tooltip }) => {
+  const activeTooltip: Partial<ITableMetricsData> =
+    tooltip && tooltip.activeData ? tooltip.activeData : {};
+  return (
+    <div
```

Comment:
Can you consider a material-ui component for consistency? `Popper` might work, but I don't know if `anchorEl` will accept an SVG element.

Reply:
I tried using Popper/Popover, however the problem is that these components create an invisible overlay, since they are using portals. So when the popover appears, the d3 is registering mouseout, thus hiding the popover immediately. There might be some way to bind to the svg elements directly, however this will mean that we will have to create individual popovers for all of the data points.

Do you have some other idea?

Reply:
Not of the top of my head, unfortunately.

Reply:
we can revisit this, and create a better framework for adding tooltips to d3 in general